File: algorithms/700-799/769. Max Chunks To Make Sorted.cpp
class Solution {
public:
    int maxChunksToSorted(vector<int>& arr) {
        if (arr.empty()) return 0;
        int currentMax = 0, res = 0;
        for (int i = 0; i < arr.size(); ++i){
            currentMax = max(currentMax, arr[i]);
            if(currentMax == i){
                res += 1;
            }
        }
        return res;
    }
};

File: algorithms/700-799/769. Max Chunks To Make Sorted.py
class Solution:
    def maxChunksToSorted(self, arr: List[int]) -> int:
        if not arr:
            return 0
        res = 0
        currentMax = -float('inf')
        for i, num in enumerate(arr):
            currentMax = max(currentMax, num)
            if currentMax == i:
                res += 1
        return res